osu.Framework/Graphics/Animations/IAnimation.cs

```diff
@@ -4,34 +4,39 @@
 namespace osu.Framework.Graphics.Animations
 {
     /// <summary>
-    /// Represents a generic, frame-based animation.
+    /// An animation / playback sequence.
     /// </summary>
     public interface IAnimation
     {
         /// <summary>
-        /// The number of frames this animation has.
+        /// The duration of the video that is being played. Can only be queried after the decoder has started decoding has loaded. This value may be an estimate by FFmpeg, depending on the video loaded.
```

Comment:
```suggestion
        /// The duration of the animation. Can only be queried after the decoder has started decoding has loaded. This value may be an estimate by FFmpeg, depending on the video loaded.
```

Also the ffmpeg blurb is a little weird to me in an interface like this, because it's not true for `Animation`.

How about making it virtual and override in Video to add the comment/possibly an exception later down the track?

Reply:
Oh, the ffmpeg part should really be removed. Now the duration is always set at a specific point in time (bdl) regardless.